Userland/Libraries/LibGfx/Matrix.h

```diff
@@ -94,6 +104,21 @@ class Matrix {
         return division;
     }
 
+    friend constexpr Matrix operator*(Matrix const& matrix, T scalar)
```

Comment:
Is the `friend` keyword needed ? I'm not even sure that it is doing something.

Also, you might want to add the `[[nodiscard]]` on operators.

Reply:
Those are not member functions (they are just friend functions placed inside the class).
 
I made the `operator*` a friend as I want both `scalar * matrix` and `matrix * scalar` to work. That first one could be a member function (but it's not for consistency), but the second one `scalar * matrix` only works as a friend function.

Reply:
Not sure it needs `[[nodiscard]]` since a statement like `foo * bar;` is obviously ignoring the result (and looks weird).

Reply:
> Those are not member functions (they are just friend functions placed inside the class).
> 
> I made the `operator*` a friend as I want both `scalar * matrix` and `matrix * scalar` to work. That first one could be a member function (but it's not for consistency), but the second one `scalar * matrix` only works as a friend function.

Got it! Thank you

Absolutely `foo * bar;` is weird, and you want the compiler to warn you about it.
That's the role of `[[nodiscard]]`

Reply:
Fair enough :smile: